x-pack/plugin/sql/qa/server/src/main/java/org/elasticsearch/xpack/sql/qa/rest/RestSqlTestCase.java

```diff
@@ -1183,4 +1205,207 @@ public static void assertResponse(Map<String, Object> expected, Map<String, Obje
             fail("Response does not match:\n" + message.toString());
         }
     }
+
+    //
+    // Async text formatting tests
+    //
+
+    public void testBasicAsyncWait() throws IOException {
+        RequestObjectBuilder builder = query("SELECT 1").waitForCompletionTimeout("1d") // wait "forever"
+            .keepOnCompletion(false);
+
+        String mode = randomMode();
+
+        Map<String, Object> expected = new HashMap<>();
+        expected.put(IS_PARTIAL_NAME, false);
+        expected.put(IS_RUNNING_NAME, false);
+        expected.put(COLUMNS_NAME, singletonList(columnInfo(mode, "1", "integer", JDBCType.INTEGER, 11)));
+        expected.put(ROWS_NAME, singletonList(singletonList(1)));
+        assertAsyncResponse(expected, runSql(builder, mode));
```

Comment:
I'm worrying again here and for the rest of the aysnc tests about flakiness. Should we use the `assertBusy` mechanism to make sure that it always works?

Reply:
The `assertBusy` could be used in this case, true (tho with extra complexity). How would you see it failing in a flaky way?

Reply:
Not easy to achieve I guess, If you want to just check locally maybe try a simple `Thread.sleep()` somewhere in the SQL chain (e.g. Analyzer), so that the execution is delayed for let's say 5secs.

Reply:
IIUC your suggestion, the (merged) local [cancellation tests](https://github.com/elastic/elasticsearch/blob/feat/sql-async/x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/CancellationTests.java) do that, they start a task and make sure it blocks at certain stages along its execution.

The tests here are all remote ones though, that are only mean to check the correct response formatting of the textual answers. The timing is irrelevant, it makes no difference between "10s" or "1d", for as long as the query finishes. But even if it doesn't finish, or the query goes wrong, it'd result in a failed test due to mismatch between expected and returned answer. That's why I wasn't sure how you see it failing in a flaky way.

Using `assertBusy` to check on task's actual completion before checking its result is possible, but don't think it's necessary. But let me know if you see it differently still.

Reply:
@bpintea Thx Bogdan, I was confused with the functionality here.
Since we have the `waitForCompletionTimeout` that ensures that we have the response (or a timeout if the test runs for 1 day :D) when we try to assert it, so no need for the `assertBusy`.